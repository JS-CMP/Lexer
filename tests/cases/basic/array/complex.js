var complexArray = [
    {
        name: "John",
        age: 30,
        details: {
            address: {
                street: "123 Main St",
                city: "New York",
                zipCode: 10001,
            },
            isActive: true,
            hobbies: ["reading", "coding", "hiking"]
        }
    },
    {
        id: 42,
        data: [
            {
                key: "nestedValue",
                value: [1, 2, { deep: "level" }]
            },
            {
                mixedTypes: [null, undefined, true, 3.14, "string", { nested: false }]
            },
        ],
    },
    {
        flag: false,
        items: [
            {
                a: [
                    {
                        b: undefined,
                    }
                ]
            }
        ]
    }
];
