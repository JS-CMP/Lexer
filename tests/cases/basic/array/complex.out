#include "types/JsAny.hpp"
#include "class/Console.hpp"
#include "types/objects/Types.hpp"

int main() {JS::Any complexArray=JS::Any(JS::Array({JS::Any(JS::Object({{"name", JS::Any("John")},{"age", JS::Any(30)},{"details", JS::Any(JS::Object({{"address", JS::Any(JS::Object({{"street", JS::Any("123 Main St")},{"city", JS::Any("New York")},{"zipCode", JS::Any(10001)},}))},{"isActive", JS::Any(true)},{"hobbies", JS::Any(JS::Array({JS::Any("reading"), JS::Any("coding"), JS::Any("hiking")}))}}))}})), JS::Any(JS::Object({{"id", JS::Any(42)},{"data", JS::Any(JS::Array({JS::Any(JS::Object({{"key", JS::Any("nestedValue")},{"value", JS::Any(JS::Array({JS::Any(1), JS::Any(2), JS::Any(JS::Object({{"deep", JS::Any("level")}}))}))}})), JS::Any(JS::Object({{"mixedTypes", JS::Any(JS::Array({JS::Any(nullptr), JS::Any(), JS::Any(true), JS::Any(3.14), JS::Any("string"), JS::Any(JS::Object({{"nested", JS::Any(false)}}))}))}})), }))},})), JS::Any(JS::Object({{"flag", JS::Any(false)},{"items", JS::Any(JS::Array({JS::Any(JS::Object({{"a", JS::Any(JS::Array({JS::Any(JS::Object({{"b", JS::Any()},}))}))}}))}))}}))}));}
